test/plugin/test_in_tail.rb

```diff
@@ -866,6 +866,46 @@ def test_log_file_without_extension
       plugin = create_driver(config, false).instance
       assert_equal expected_files, plugin.expand_paths.sort
     end
+
+    # For https://github.com/fluent/fluentd/issues/1455
+    # This test is fragile because test content depends on internal implementaion.
+    # So if you modify in_tail internal, this test may break.
+    def test_unwatched_files_should_be_removed
+      config = config_element("", "", {
+                                "tag" => "tail",
+                                "path" => "#{TMP_DIR}/*.txt",
+                                "format" => "none",
+                                "pos_file" => "#{TMP_DIR}/tail.pos",
+                                "read_from_head" => true,
+                                "refresh_interval" => 1,
+                              })
+      d = create_driver(config, false)
+      d.run(expect_emits: 1, shutdown: false) do
+        File.open("#{TMP_DIR}/tail.txt", "ab") { |f| f.puts "test3\n" }
+      end
+
+      assert_equal 1, d.instance.instance_variable_get(:@tails).keys.size
+      File.unlink("#{TMP_DIR}/tail.txt")
+      sleep 2
```

Comment:
This should be `waiting(5){ sleep 0.1 until the_size_of_tails == 0 }` instead of just `sleep 2` for stability.